pkg/apiserver/sync/worker.go

```diff
@@ -58,8 +58,14 @@ func startAppSyncing(ctx context.Context, factory dynamicinformer.DynamicSharedI
 	informer := factory.ForResource(v1beta1.SchemeGroupVersion.WithResource("applications")).Informer()
 	getApp := func(obj interface{}) *v1beta1.Application {
 		app := &v1beta1.Application{}
-		bs, _ := json.Marshal(obj)
-		_ = json.Unmarshal(bs, app)
+		bs, err := json.Marshal(obj)
+		if err != nil {
+			logrus.Fatal(err)
```

Comment:
```suggestion
			klog.Errorf("failed to convert obj %v to application with marshal err %v", obj, err)
			return nil
```

the same below, some places we shouldn't panic the process, just report the log is fine.

Reply:
Thanks @wonderflow !